bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/Desugar.scala

```diff
@@ -187,6 +187,25 @@ object Desugar extends Transformer with TreeDSL {
       treeCopy.MatchStatement(expr, value,
           clauses map(c => matchExpressionClauseToBindStatement(result, c)),
           exprToBindStatement(result, elseExpression))
+
+    case Record(label, fields) =>
+      val lastI = fields.lastIndexWhere(_.value match {
+        case CallExpression(callable, args) => true
+        case _                              => false
+      })
```

Comment:
Ok, now I got the book. It is not discussed in it, but I thought about it. We could unwrap all the fields behind the call. What matters, however is that the arguments will be evaluated from left to right, because of semantics. And that is not ensured by this current method.
So we could unwrap all the fields, but it would be heavy in variables and statements. What I propose is to defer only the last field if it is a call, it seems a good compromise between correctness and efficiency.